src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,37 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
+            int length = Interop.GlobalizationNative.ToAscii(flags, unicode, count, null, 0);
```

Comment:
Thanks.

@hughbe, I may have missed it in the discussion, but did you consider doing this only for the stackalloc case?  If the goal here is to avoid unnecessary allocations, if count is less than StackAllocThreshold, we can do the call once using the stack allocation; if it's enough, we won't have to make a second call and won't have allocated any heap memory, and if it's not enough, we'll still only make two calls and allocate on the heap the correct amount.  We can do the two-call behavior then when the initial count exceeds the StackAllocThreshold.  That way, for short strings we get the best of both worlds (no extra allocation, one call), just as we do today, and for long strings, we make two calls and have no unnecessary allocation.